src/desktop/settings.rs

```diff
@@ -186,6 +187,34 @@ impl<'a> Settings<'a> {
             .await
     }
 
+    /// Listen to changes of the system's preferred color scheme
+    ///
+    /// The stream will also initially yield the current color scheme preference and filter out
+    /// duplicates.
+    pub async fn receive_color_scheme_changed_with_init(
```

Comment:
So two questions here:

- What is really the use case of yielding the current color scheme?

- What does filtering out the duplicates mean? how can the signal be emitted if the setting didn't change, it should not. If it happens, it is a backend issue and should not be worked around from a frontend library.

Reply:
> What is really the use case of yielding the current color scheme?

The way I would use it in my cli application is waiting for a changed color-scheme and perform an action when it changes. Not sure how it would be used in a bigger (gui) application.


> What does filtering out the duplicates mean? how can the signal be emitted if the setting didn't change, it should not. If it happens, it is a backend issue and should not be worked around from a frontend library.

That's a good question. I just experienced that on my machine the signal always gets emitted twice. A google search wasn't helpful. But probably you're right and we shouldn't handle this issue here.

Reply:
Which xdg-desktop-portal you have installed? Also could you use something like Bustle, record the dbus traffic and share the pcap file here so we can investigate which part of the stack is not behaving as expected?

Regarding the cli app you are working on, I think the best would be to have a closure and call it on the initial state as well as when the state changes. That would be much cleaner then the current API suggestion.

Reply:
> Which xdg-desktop-portal you have installed? Also could you use something like Bustle, record the dbus traffic and share the pcap file here so we can investigate which part of the stack is not behaving as expected?

Thank you, that you want to investigate this case. I'm using Ubuntu 22.04 but I'm not sure which backend is used. Both gnome.portal  gtk.portal support the "Settings" portal and are `UseIn=gnome`, `$XDG_SESSION_DESKTOP` is `ubuntu`. So I guess it would be (because it comes first in the alphabet) gnome.portal? Here the [pcap]([2023-08-11 15:27:03.zip](https://github.com/bilelmoussaoui/ashpd/files/12321956/2023-08-11.15.27.03.zip)

> Regarding the cli app you are working on, I think the best would be to have a closure and call it on the initial state as well as when the state changes. That would be much cleaner then the current API suggestion.

Yes that's definitely doable.

Reply:
Looking at the pcap it could be that both gtk and gnome are active, which might be the root cause. But I don't know how I could have misconfigured my system. Maybe because I installed additional desktop environments? It's also not crucial to fix this on my machine..

Reply:
> Looking at the pcap it could be that both gtk and gnome are active, which might be the root cause. But I don't know how I could have misconfigured my system. Maybe because I installed additional desktop environments? It's also not crucial to fix this on my machine..

You can have multiple backends, each of them providing only a subset of the portals. It is normal to have both gtk & gnome ones.